rpcserver.go

```diff
@@ -7034,6 +7034,92 @@ func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
 	case len(firstHopRecords) > 0:
 		// Continue below.
 
+	case req.RfqId != nil:
+		// Check if the provided rfq ID matches the expected length.
+		if len(req.RfqId) != 32 {
+			return fmt.Errorf("rfq id length must be 32")
+		}
+
+		// Now let's try to perform an internal lookup to see if there's
+		// an actual quote on this ID.
+		var rfqID rfqmsg.ID
+		copy(rfqID[:], req.RfqId)
+
+		var quote rfqmsg.SellAccept
```

Comment:
nit: can have this as a pointer, then below we can do a `nil` check instead of comparing the ID against an empty ID.